### PR TITLE
Feature/remember me

### PR DESCRIPTION
**Pull request by Sinisa198**

### CODE REVIEW

File: moja-kuhinjica-frontend/src/components/label/ErrorLabel.tsx [hunk 1]
```diff
@@ -1,12 +1,37 @@
...
+    const [email, setEmail] = useState<string>('')
+    const [afterEmail, setAfterEmail] = useState<string>('')
+    const regex = /(.*) email (.*?) (.*?)$/
+
+    useEffect(() => {
+        const matches = content?.match(regex)
+        if (matches?.length === COMAPRING_NUMBER) {
+            setBeforeEmail(matches[1])
+            setEmail(matches[2])
+            setAfterEmail(matches[3])
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/components/label/ErrorLabel.tsx:26]: I've commented this here
https://github.com/Positive-Tech/moja-kuhinjica-FE/pull/97

File: moja-kuhinjica-frontend/src/components/modal/CanCancelReservation.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                </label>
+                <button
+                    className="modalContainer__formContainer__formDiv__formTitle__formButton"
+                    onClick={closeModal}
+                >
+                    {t(buttonText)}
+                </button>
+            </div>
+        )
+    }
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/components/modal/CanCancelReservation.tsx:42]: Commented here
https://github.com/Positive-Tech/moja-kuhinjica-FE/pull/97

File: moja-kuhinjica-frontend/src/components/modal/CanCancelReservation.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            isOpen={modalIsOpen}
+            onRequestClose={closeModal}
+            style={bgModal}
+            className="modalContainer"
+            ariaHideApp={false}
+        >
+            <div className="modalContainer__formContainer">
+                <div className="modalContainer__formContainer__formDiv">
+                    <label className="modalContainer__formContainer__formDiv__formTitle">
+                        {getBody()}
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/components/modal/CanCancelReservation.tsx:55]: Here https://github.com/Positive-Tech/moja-kuhinjica-FE/pull/97

File: moja-kuhinjica-frontend/src/components/modal/disabledReservation/DisabledReservationModal.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                </label>
+                <button
+                    className="modalContainer__formContainer__formDiv__formTitle__formButton"
+                    onClick={closeModal}
+                >
+                    {t(buttonText)}
+                </button>
+            </div>
+        )
+    }
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/components/modal/disabledReservation/DisabledReservationModal.tsx:42]: Here https://github.com/Positive-Tech/moja-kuhinjica-FE/pull/97

File: moja-kuhinjica-frontend/src/components/modal/disabledReservation/DisabledReservationModal.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            isOpen={modalIsOpen}
+            onRequestClose={closeModal}
+            style={bgModal}
+            className="modalContainer"
+            ariaHideApp={false}
+        >
+            <div className="modalContainer__formContainer">
+                <div className="modalContainer__formContainer__formDiv">
+                    <label className="modalContainer__formContainer__formDiv__formTitle">
+                        {getBody()}
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/components/modal/disabledReservation/DisabledReservationModal.tsx:55]: Here https://github.com/Positive-Tech/moja-kuhinjica-FE/pull/97

File: moja-kuhinjica-frontend/src/components/modal/login/LoginModal.tsx [hunk 6]
```diff
@@ -102,6 +108,17 @@ export const LoginModal = ({
                             required: t('Šifra je obavezna.'),
                         }}
                     />
+                    <div className="remember-me">
+                        <input
+                            id="cb1"
+                            checked={rememberMe}
+                            onChange={() => {
+                                setRememberMe(!rememberMe)
+                            }}
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/components/modal/login/LoginModal.tsx:117]: Here https://github.com/Positive-Tech/moja-kuhinjica-FE/pull/97

File: moja-kuhinjica-frontend/src/components/modal/login/LoginModal.tsx [hunk 7]
```diff
@@ -102,6 +108,17 @@ export const LoginModal = ({
...
+                            type="checkbox"
+                        />
+                        <p>Zapamti me</p>
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/components/modal/login/LoginModal.tsx:120]: Here https://github.com/Positive-Tech/moja-kuhinjica-FE/pull/97

**Review by vladavoX — CHANGES_REQUESTED** [review]
Please change the source branch. If you are checking out to a new branch from a certain branch, PR needs to be back in that branch, so I don't see double code that I've already commented

**Review by kiljee — APPROVED** [review]

File: moja-kuhinjica-frontend/src/components/label/ErrorLabel.tsx [hunk 8]
```diff
@@ -1,12 +1,37 @@
...
+
 export const ErrorLabel = ({
     content,
     style,
 }: IErrorLabelProps): JSX.Element => {
-    return <label className={`errorLabel ${style}`}>{content}</label>
+    const { t } = useTranslation()
+    const [beforeEmail, setBeforeEmail] = useState<string>('')
+    const [email, setEmail] = useState<string>('')
+    const [afterEmail, setAfterEmail] = useState<string>('')
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/components/label/ErrorLabel.tsx:18]: create single state

File: moja-kuhinjica-frontend/src/pages/email/password/[token].tsx [hunk 9]
```diff
@@ -8,8 +8,10 @@ import { useRouter } from 'next/router'
 import UserService from '@/service/User.service'
 import { Oval } from 'react-loader-spinner'
 import { routes } from '@/constants/constants'
+import { useTranslation } from 'react-i18next'
 
 const ResetPasswordPage = (): JSX.Element => {
+    const { t } = useTranslation()
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/email/password/[token].tsx:14]: use this `t` to translate

File: moja-kuhinjica-frontend/src/pages/login/index.tsx [hunk 10]
```diff
@@ -81,6 +82,17 @@ const LoginPage = (): JSX.Element => {
                         }}
                         style="loginContainer__wrapper__formDiv__input"
                     />
+                    <div className="remember-me">
+                        <input
+                            id="cb1"
+                            checked={rememberMe}
+                            onChange={() => {
+                                setRememberMe(!rememberMe)
+                            }}
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/login/index.tsx:91]: inline `onChange`

File: moja-kuhinjica-frontend/src/pages/login/index.tsx [hunk 11]
```diff
@@ -81,6 +82,17 @@ const LoginPage = (): JSX.Element => {
...
+                            type="checkbox"
+                        />
+                        <p>Zapamti me</p>
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/login/index.tsx:94]: translate

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 12]
```diff
@@ -40,11 +42,13 @@ import { routes } from '../../constants/constants'
...
 const HEADER_TYPE = 'red'
 const INITIAL_MEAL_AMOUNT = 1
+const DISABLED_MESSAGE = 'Ne možete da rezervišete posle 10 ujutru'
 const RESERVATION_SUCCESS = 'Rezervacija je uspešna'
 const RESERVATION_FAIL = 'Neuspešna rezervacija'
 const RESERVATION_SUCCESS_MESSAGE =
     'Vaša rezervacija je sačuvana. Možete je pogledati na stranici'
-const EMPTY_CART_MESSAGE = 'Vaša korpa je prazna, ukoliko želite da pogledate svoje rezervacije možete otići na stranicu'
+const EMPTY_CART_MESSAGE =
+    'Vaša korpa je prazna, ukoliko želite da pogledate svoje rezervacije možete otići na stranicu'
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:51]: move these to seprate file

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 13]
```diff
@@ -136,6 +142,10 @@ const MealReservation = (): JSX.Element => {
             })
     }
     const addToCart = (meal: IMeal): void => {
+        if (!isBookingAllowed(activeDate)) {
+            return setShowDisabledReservation(true)
+        }
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:147]: write this inline

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 14]
```diff
@@ -237,6 +247,13 @@ const MealReservation = (): JSX.Element => {
                 }}
                 buttonText="OK"
             />
+            <DisabledReservationModal
+                modalIsOpen={showDisabledReservation}
+                closeModal={() => setShowDisabledReservation(!true)}
+                title={DISABLED_MESSAGE}
+                buttonText={'OK'}
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:254]: buttonText='OK'

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 15]
```diff
@@ -275,11 +292,11 @@ const MealReservation = (): JSX.Element => {
...
                                         active={active === activeTabIndex}
                                         onClick={() => {
-                                            setDeyOfWeek(date.day())
-                                            setActive(activeTabIndex)
                                             if (cartItems.length) {
                                                 handleTabClickWithCartItems()
                                             } else {
+                                                setDeyOfWeek(date.day())
+                                                setActive(activeTabIndex)
                                                 setActiveDate(day.date)
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:300]: dont use `else`, write two `if`s that return

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 16]
```diff
@@ -342,12 +360,17 @@ const MealReservation = (): JSX.Element => {
...
+                                        <span
                                             className="mealReservation__container__menuDiv__cartContainer__cartWrapper__emptyCartDiv__emptyCartLabel__emptySpan"
-                                            onClick={() => router.push(routes.MY_RESERVATIONS_PAGE)}    
-                                        > 
+                                            onClick={() =>
+                                                router.push(
+                                                    routes.MY_RESERVATIONS_PAGE
+                                                )
+                                            }
+                                        >
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:373]: use Link

File: moja-kuhinjica-frontend/src/pages/myReservations/index.tsx [hunk 17]
```diff
@@ -21,33 +21,18 @@ import { RegularButton } from '@/components/button/RegularButton'
...
-                    isActive
-                    onClick={() => router.push(routes.MEAL_RESERVATION_PAGE)}
-            />
-        </div>
-    )
-}
+const NoReservationsMessage: React.FC = () => (
+    <div className="myReservationsPage__colDiv__rowDiv">
+        <label className="myReservationsPage__colDiv_an_rowDiv__infoLabel">
+            Nema rezervacija za ovaj datum.
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/myReservations/index.tsx:32]: translate

**Review by vladavoX — CHANGES_REQUESTED** [review]

**Review by vladavoX — APPROVED** [review]

**Pull request merged**